packages/cli/src/tools/addNodeLinker.ts

```diff
@@ -0,0 +1,17 @@
+import path from 'path';
+import fs from 'fs';
+
+/**
+ Creates a `.yarnrc.yml` file in the root of the project to force Yarn to use the `node-modules` linker, because React Native doesn't support the Plug'n'Play node linker.
+ */
+
+const addNodeLinker = (root: string) => {
+  const yarnrcFileContent = 'nodeLinker: node-modules\n';
+
+  fs.writeFileSync(path.join(root, '.yarnrc.yml'), yarnrcFileContent, {
```

Comment:
I think this will force users onto Yarn v3. I don't think we want to do that. Can we instead try to detect the version and suggest solutions?

Reply:
I did some testing locally and it looks like just adding a `.yarnrc.yml` won't trigger Yarn to use v3, which is good. But I still think we want to avoid any confusion having this extra file may introduce.

Reply:
Yeah, basically:
- If Yarn Classic, all is good.
- Else check if `.yarnrc.yml` exists and whether `nodeLinker` is correctly set (and suggest/update if not)

If you're using Yarn v3, you can assume that `.yarnrc.yml` exists as you are required to specify the path to Yarn. You can find an example here: https://github.com/microsoft/react-native-test-app/blob/trunk/.yarnrc.yml